plugin/plugins/filesystem/nodes/pinfs.go

```diff
@@ -0,0 +1,155 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+	gopath "path"
+
+	"github.com/hugelgupf/p9/p9"
+	"github.com/hugelgupf/p9/unimplfs"
+	nodeopts "github.com/ipfs/go-ipfs/plugin/plugins/filesystem/nodes/options"
+	fsutils "github.com/ipfs/go-ipfs/plugin/plugins/filesystem/utils"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	coreoptions "github.com/ipfs/interface-go-ipfs-core/options"
+)
+
+var _ p9.File = (*PinFS)(nil)
+var _ fsutils.WalkRef = (*PinFS)(nil)
+
+type PinFS struct {
+	unimplfs.NoopFile
+	p9.DefaultWalkGetAttr
+
+	IPFSBase
+	ents []p9.Dirent
+}
+
+func PinFSAttacher(ctx context.Context, core coreiface.CoreAPI, ops ...nodeopts.AttachOption) p9.Attacher {
+	pd := &PinFS{IPFSBase: newIPFSBase(ctx, "/pinfs", core, ops...)}
+	pd.qid.Type = p9.TypeDir
+	pd.meta.Mode, pd.metaMask.Mode = p9.ModeDirectory|IRXA, true
+
+	// set up our subsystem, used to relay walk names to IPFS
+	subOpts := []nodeopts.AttachOption{
+		nodeopts.Parent(pd),
+		nodeopts.Logger(logging.Logger("IPFS")),
+	}
+
+	subsystem, err := IPFSAttacher(ctx, core, subOpts...).Attach()
+	if err != nil {
+		panic(err)
+	}
+
+	pd.proxy = subsystem.(fsutils.WalkRef)
+
+	return pd
+}
+
+// this root has no paths to walk; forking anythign besides the fs doesn't make sense for us
```

Comment:
This is wrong.
We use the operations context and need to fork it, otherwise a close to any reference will cancel the context for all of them.
KeyFS also does this.

Add tests:
take a reference, make 2 clones of it, close one of them, open the other, this should succeed.